pkg/inject/oauth-proxy.go

```diff
@@ -56,14 +60,19 @@ func getOAuthProxyContainer(jaeger *v1.Jaeger) corev1.Container {
 
 	commonSpec := util.Merge([]v1.JaegerCommonSpec{jaeger.Spec.Ingress.JaegerCommonSpec, jaeger.Spec.JaegerCommonSpec})
 
+	volumeMounts := []corev1.VolumeMount{{
+		MountPath: "/etc/tls/private",
+		Name:      service.GetTLSSecretNameForQueryService(jaeger),
+	}}
+
+	// we can only get VolumeMounts from the top-level node
+	volumeMounts = append(volumeMounts, jaeger.Spec.JaegerCommonSpec.VolumeMounts...)
```

Comment:
Problem is that these mounts would be added to all components. The IngressSpec (or OpenShift specific node) probably needs to support its own volumes/mounts - and need to consider whether any other part of the `CommonSpec` is relevant - so should it be used or just the volume/mounts?

Reply:
I left it very simple at the moment, because I think we can discuss it more broadly. We should include storage-related objects in the mix as well, like cronjobs.

I changed the code slightly to mount only when needed, ie, when the `htpasswd-file` option is set.

Reply:
I was thinking more about the other components which will just mount the volumes listed at the top level :)

Reply:
I see now what you mean. Yes, in order to have htpasswd, one would have to specify it at the top-level, which will cause all other components to mount the htpasswd secret. For this initial implementation, I think it's OK, but we do need to come up with a proper solution, also taking into consideration the other common spec values. Should we do this in a follow-up PR?

Reply:
Yes a separate PR should be fine.

Reply:
#574